config/initializers/content_security_policy.rb

```diff
@@ -12,7 +12,8 @@
   policy.object_src  :none
   policy.script_src  :self, "https://secure.gaug.es", "https://www.fastly-insights.com"
   policy.style_src   :self, "https://fonts.googleapis.com"
-  policy.connect_src :self, "https://s3-us-west-2.amazonaws.com/rubygems-dumps/", "https://*.fastly-insights.com", "https://fastly-insights.com", "https://api.github.com"
+  policy.connect_src :self, "https://s3-us-west-2.amazonaws.com/rubygems-dumps/", "https://*.fastly-insights.com", "https://fastly-insights.com",
+    "https://api.github.com", "http://localhost:*"
```

Comment:
So I looked into CSPs a bit more. The gist of this change is that we allow our script to connect to `localhost` on a random port in order to complete the CLI flow.

Is this a security risk? Some light searching reveals that folks [don't think so](https://stackoverflow.com/questions/43742261/is-it-unsafe-to-add-localhost-to-content-security-policy). Adding to this, the risk is that an attacker can inject content into some rubygems.org content that, when served in HTML, causes their browser to perform an HTTP action via a crafted `<a>` element or one of the various HTTP APIs in JavaScript (eg `fetch`).

The problem is that the attacker is unlikely to have control of `localhost`. If they've managed to set up an HTTP process on `localhost` then they can take more direct action against the user, such as credential collection.

I don't think this represents a meaningful risk.

Reply:
Thanks so much for looking into this! That's the conclusion @jenshenny and I came to as well when we were pairing. Having a third set of 👀 on it makes me feel much more comfortable moving forward with this solution 🎉

Reply:
does it potentially make sense to scope the CSP change to the one controller that is making the request to localhost? https://api.rubyonrails.org/classes/ActionController/ContentSecurityPolicy/ClassMethods.html#method-i-content_security_policy